src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/ClientCreds/JsonWebToken.cs

```diff
@@ -212,6 +220,21 @@ public JWTHeaderWithCertificate(IClientAssertionCertificate credential)
                 : base(credential)
             {
                 _thumbPrint = this.Credential.Thumbprint;
+                X509CertificatePublicCertValue = string.Empty;
+
+                //Check to see if credential is our implementation or developer provided.
+                if (!credential.GetType().ToString().Contains("Microsoft.IdentityModel"))
+                {
+                    CallState.Default.Logger.Warning(null, "The implementation of IClientAssertionCertificate is developer provided and it should be replaced with library provided implmentation.");
+                    return;
+                }
+
+#if (NET45 || NETSTANDARD1_3)
+                if (credential is ClientAssertionCertificate cert)
+                {
+                    X509CertificatePublicCertValue = Base64UrlEncoder.Encode(cert.Certificate.ToString());
```

Comment:
I thought you mentioned doing Base64 encoding, but I see Base64Url encoding being used instead. This will not work.

Reply:
ToString() function does not return the actual public certificate. There already code in ADAL that pulls and encodes certificate data. please dig through the code and find the right API calls